### DISCUSSION TITLE
Is there an `Enum` for function codes?

### DISCUSSION THREAD

**Discussion by jamesbraza** [pymodbus-dev/pymodbus#1657] [Q&A] [answered]
I see within `pymodbus==3.3.2`, some `ModbusResponse` subclasses define `function_code` and `function_code_name`, like [here](https://github.com/pymodbus-dev/pymodbus/blob/5a9f083398b1ecc8b4c3c4ebf13d636c42f55151/pymodbus/register_read_message.py#L192).

Is there an `Enum` or some central place that maps between function codes and readable names?

**Comment by janiversen** [comment, marked as answer]
There are no enum, but there is a list with all function codes and the corresponding class.

**Reply by jamesbraza** [reply]
Where can one find the list?

I was looking for it, failed to find it

**Reply by janiversen** [reply]
factory.py, but be aware it is internal and will probably change soon.
